pkg/sql/schemachanger/scexec/exec_mutation.go

```diff
@@ -142,6 +156,81 @@ func executeDescriptorMutationOps(ctx context.Context, deps Dependencies, ops []
 	return b.ValidateAndRun(ctx)
 }
 
+func eventLogEntriesForStatement(statementEvents []eventPayload) (logEntries []eventPayload) {
+	// A dependent event is one which is generated because of a
+	// dependency getting modified from the source object. An example
+	// of this is a DROP TABLE will be the source event, which will track
+	// any dependent views dropped.
+	var dependentEvents = make(map[uint32][]eventPayload)
+	var sourceEvents = make(map[uint32]eventPayload)
+	// First separate out events, where the first event generated will always
+	// be the source and everything else before will be dependencies if they have
+	// the same subtask ID.
+	for _, event := range statementEvents {
+		dependentEvents[event.metadata.SubWorkID] = append(dependentEvents[event.metadata.SubWorkID], event)
+	}
+	// Split of the source events.
+	orderedSubWorkID := make([]uint32, 0, len(dependentEvents))
+	for subWorkID := range dependentEvents {
+		elems := dependentEvents[subWorkID]
+		sort.SliceStable(elems, func(i, j int) bool {
+			return elems[i].metadata.SourceElementID < elems[j].metadata.SourceElementID
+		})
+		sourceEvents[subWorkID] = elems[0]
+		dependentEvents[subWorkID] = elems[1:]
+		orderedSubWorkID = append(orderedSubWorkID, subWorkID)
+	}
+	// Store an ordered list of sub-work IDs for deterministic
+	// event order.
+	sort.SliceStable(orderedSubWorkID, func(i, j int) bool {
+		return orderedSubWorkID[i] < orderedSubWorkID[j]
+	})
+	// Collect the dependent objects for each
+	// source event, and generate an event log entry.
+	for _, subWorkID := range orderedSubWorkID {
+		// Determine which objects we should collect.
+		collectDependentViewNames := false
+		collectDependentSchemaNames := false
+		sourceEvent := sourceEvents[subWorkID]
+		switch sourceEvent.event.(type) {
+		case *eventpb.DropDatabase:
+			// Drop database only reports dependent schemas.
```

Comment:
This feels so arbitrary. Is there some principle here? Do all dropped descriptors end up getting captured but I'm just missing how? Maybe printing the complete events in the end-to-end test will make it more obvious.

Reply:
I have a feeling that Faizan is working on this right now.